File: app/src/main/java/com/legue/axel/lolsummonertool/database/converter/DoubleListConverters.java
package com.legue.axel.lolsummonertool.database.converter;

import androidx.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.Collections;
import java.util.List;

public class DoubleListConverters {
    @TypeConverter
    public static List<Double> doubleToList(String value) {
        Gson gson = new Gson();

        if (value == null) {
            return Collections.emptyList();
        }

        Type listType = new TypeToken<List<Float>>() {
        }.getType();

        return gson.fromJson(value, listType);
    }

    @TypeConverter
    public static String ListToString(List<Double> descriptions) {
        Gson gson = new Gson();
        return gson.toJson(descriptions);
    }
}

File: app/src/main/java/com/legue/axel/lolsummonertool/database/converter/DoubleListConverters.kt
package com.legue.axel.lolsummonertool.database.converter

import androidx.room.TypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

class DoubleListConverters {
    @TypeConverter
    fun doubleToList(value: String?): List<Double>? {
        if (value == null || value == "null") {
            return emptyList()
        }

        val listType = object : TypeToken<List<Double>>() {}.type
        val objects = Gson().fromJson(value, listType) as List<Double>
        return objects.toList()
    }

    @TypeConverter
    fun ListToString(descriptions: List<Double>): String {
        return Gson().toJson(descriptions)
    }
}
